deployment/helm/templates/controller.yaml

```diff
@@ -36,17 +36,27 @@ spec:
       imagePullSecrets:
         {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- if .Values.controller.linuxOnly }}
-      nodeSelector:
-        "kubernetes.io/os": linux
-      {{- end }}
-      {{- if .Values.controller.onlyOnMaster }}
-      # Only run this pod on the master.
+      {{- if .Values.controller.allowOnControlPlane }}
       tolerations:
         # Allow this pod to run on the master.
         - key: node-role.kubernetes.io/master
           effect: NoSchedule
+      {{- end }}
+      {{- if or .Values.controller.linuxOnly .Values.controller.onlyOnControlPlane }}
```

Comment:
Should this if not include .Values.controller.nodeSelectors?

Reply:
Or should nodeSelectors not be here given that they are also below?

Reply:
there are 2 clauses here.  one for when either linuxOnly or onlyOnControlPlane ... and one when neither of those.

in both clauses, nodeSelectors is added if it is set.

this allows us to only include the yaml `nodeSelector` iff there is something to add to nodeSelector.

Reply:
a simpler form would be:

```yaml
      {{- if or .Values.controller.linuxOnly .Values.controller.onlyOnMaster .Values.controller.nodeSelectors }}
      nodeSelector:
        {{- if .Values.controller.nodeSelectors }}
          {{- toYaml .Values.controller.nodeSelectors | nindent 8 }}
        {{- end }}
        {{- if .Values.controller.linuxOnly }}
        "kubernetes.io/os": linux
        {{- end }}
        {{- if .Values.controller.onlyOnMaster }}
        node-role.kubernetes.io/master: ""
        {{- end }}
      {{- end }}
```